lib/logging/include/SWO.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+namespace Logger {
+
+// C-style printf - prints to SWO interface
+void SWO_printf(const char* format, ...);
+
+// SWO logger
+class SWO {
+ public:
+  SWO& operator<<(long n);
```

Comment:
Can you template these?

Reply:
Actually is there a clean way to do it with templates? Since this same function is overloaded with signed, unsigned, floats, and strings

Reply:
Not sure for sure but I think so

Reply:
I was having trouble templating this, it wouldn't let me do template specialization with the member functions.. Can I leave the overloads for now?